lib/ret_web/channels/hub_channel.ex

```diff
@@ -10,6 +10,7 @@ defmodule RetWeb.HubChannel do
     Hub
     |> Repo.get_by(hub_sid: hub_sid)
     |> Repo.preload(scene: [:model_owned_file, :screenshot_owned_file])
+    |> Hub.ensure_valid_entry_code!()
```

Comment:
I wonder if we should guard against entry codes that are just about to expire. i.e. If an entry code is about to expire in less than 5 minutes, generate a new one before sending it to the user. And ideally the entry code should be generated when the user decides to share it, not when they create/enter a room

Maybe the current behavior is good enough for v1

Reply:
Ah yeah, I built this version with the assumption that these codes are only relevant for like the first hour or so of a room, so didn't worry too much about race conditions or boundary conditions around the expiration time. I think probably what you want is something where we generate a new code before the old one expires with some time window, so old links continue to work but new entrants get new codes. I think this can be an iteration later if this stuff ends up working well.